subsys/net/l2/canbus/canbus.c

```diff
@@ -1,49 +1,1333 @@
 /*
- * Copyright (c) 2019 Intel Corporation.
+ * Copyright (c) 2019 Alexander Wachter.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <logging/log.h>
-LOG_MODULE_REGISTER(net_l2_canbus, LOG_LEVEL_NONE);
+LOG_MODULE_REGISTER(net_l2_canbus, CONFIG_NET_L2_CANBUS_LOG_LEVEL);
 
 #include <net/net_core.h>
 #include <net/net_l2.h>
 #include <net/net_if.h>
 #include <net/net_pkt.h>
-#include <net/socket_can.h>
+#include <net/can.h>
+#include "canbus_internal.h"
+#include <6lo.h>
+#include <timeout_q.h>
+#include <string.h>
+#include <misc/byteorder.h>
+#include <net/ethernet.h>
+#include <net/net_ip.h>
+#include <string.h>
 
-static inline enum net_verdict canbus_recv(struct net_if *iface,
-					   struct net_pkt *pkt)
+#define NET_CAN_WFTMAX 2
+#define NET_CAN_ALLOC_TIMEOUT K_MSEC(100)
+
+/* Minimal separation time betwee frames */
+#define NET_CAN_STMIN CONFIG_NET_L2_CANBUS_STMIN
+#define NET_CAN_BS CONFIG_NET_L2_CANBUS_BS
+
+#define NET_CAN_DAD_SEND_RETRY 5
+#define NET_CAN_DAD_TIMEOUT K_MSEC(100)
+
+extern u16_t net_calc_chksum(struct net_pkt *pkt, u8_t proto);
+
+static struct canbus_l2_ctx l2_ctx;
+
+static struct k_work_q net_can_workq;
+K_THREAD_STACK_DEFINE(net_can_stack, 512);
+
+char *net_sprint_addr(sa_family_t af, const void *addr);
+
+#if CONFIG_NET_L2_CANBUS_LOG_LEVEL >= LOG_LEVEL_DBG
+static void print_ip_hdr(struct net_ipv6_hdr *ip_hdr)
```

Comment:
For all local function, add canbus_ prefix

Reply:
Done.